### PR TITLE
Add API readiness checklist to Release Issue and Review PR

### PR DESCRIPTION
**Pull request by hdamker**
#### What type of PR is this?

* enhancement/feature

#### What this PR does / why we need it:

Integrates the API readiness checklist into the automated release workflow at two touchpoints:

**Release Issue** — preparation prerequisites checklist with readiness matrix (collapsible), shown in PLANNED state so codeowners can verify before `/create-snapshot`

**Release Review PR** — status-aware review checklists for codeowners and release management:
- Codeowner Review: separated into "Update this PR" (CHANGELOG) and "Confirm readiness" (status-specific items following the M/O matrix)
- Release Management Review: CHANGELOG rules, SemVer compliance, and readiness asset confirmation with the full readiness matrix as reference
- Snapshot verification section (temporary, for automation introduction phase)

See the templates directly:
- [`release_issue.mustache`](https://github.com/hdamker/tooling/blob/api-readiness-checklist/release_automation/templates/issue_bodies/release_issue.mustache) — Release Issue body
- [`release_review_pr.mustache`](https://github.com/hdamker/tooling/blob/api-readiness-checklist/release_automation/templates/pr_bodies/release_review_pr.mustache) — Release Review PR body

Other changes:
- Per-API status label (alpha / rc / initial public / stable public) derived from target status and version, shown in PR API table
- Bot message wording aligned to "Update, review, and merge Release PR"
- Release issue body externalized from code into Mustache template
- Extended documentation in `release_automation/docs/api-readiness-checklist.md` (Terms, verification hints, review process, legacy mapping) — related to camaraproject/ReleaseManagement#389

#### Which issue(s) this PR fixes:

Part of the release automation readiness integration.
Fixes https://github.com/camaraproject/ReleaseManagement/issues/368

#### Special notes for reviewers:

The readiness matrix (M/O per API status) is embedded as a convenience copy in both templates. The canonical source will be `documentation/readiness/api-readiness-checklist.md` in camaraproject/ReleaseManagement (see PR camaraproject/ReleaseManagement#389 for that discussion).

#### Changelog input

```
 release-note
Add API readiness checklist integration to Release Issue and Review PR templates
```

#### Additional documentation

This section can be blank.

```
docs
Extended readiness documentation in release_automation/docs/api-readiness-checklist.md
```

### CODE REVIEW

File: release_automation/docs/api-readiness-checklist.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+## Purpose
+
+Before an API repository can be released, codeowners must ensure that certain assets are in place and meet quality expectations appropriate for the declared target API status. This checklist:
+
+- Defines the required release assets and their expected locations
+- Specifies which assets are mandatory (M) or optional (O) per API status level
+- Clarifies the division of responsibility: **codeowners prepare** the assets, **automation validates** what it can, and **release management reviewers** verify the rest
+
+This document replaces the deprecated per-API `API-Readiness-Checklist.md` files. Readiness is now tracked through `release-plan.yaml` configuration, preparation prerequisites in the Release Issue, and the review checklist in the Release PR.
```
[the remark this answers is not in the document]
Comment (tanjadegroot) [reply release_automation/docs/api-readiness-checklist.md]: I would not refer to the past approach in the new documentation.  New people do not know about the previous process.

File: release_automation/templates/bot_messages/command_rejected.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
 **❌ Command rejected: `/{{command}}` — State: `{{state}}`**
 {{error_message}}
 
-**Valid actions:**{{#state_snapshot_active}}<br>→ **Merge the Release PR to create a draft release**<br>→ `/discard-snapshot <reason>` — discard and return to `planned`{{/state_snapshot_active}}{{#state_draft_ready}}<br>→ **`/publish-release --confirm {{release_tag}}` — publish the release**<br>→ `/delete-draft <reason>` — delete draft and return to `planned`{{/state_draft_ready}}{{^state_snapshot_active}}{{^state_draft_ready}}<br>→ **`/create-snapshot` — create a release snapshot**{{/state_draft_ready}}{{/state_snapshot_active}}
+**Valid actions:**{{#state_snapshot_active}}<br>→ **Update, review, and merge the Release PR to create a draft release**<br>→ `/discard-snapshot <reason>` — discard and return to `planned`{{/state_snapshot_active}}{{#state_draft_ready}}<br>→ **`/publish-release --confirm {{release_tag}}` — publish the release**<br>→ `/delete-draft <reason>` — delete draft and return to `planned`{{/state_draft_ready}}{{^state_snapshot_active}}{{^state_draft_ready}}<br>→ **`/create-snapshot` — create a release snapshot**{{/state_draft_ready}}{{/state_snapshot_active}}
```
Comment (tanjadegroot) [inline release_automation/templates/bot_messages/command_rejected.md]:
```suggestion
**Valid actions:**{{#state_snapshot_active}}<br>→ **Update, review, and merge the Release PR to create a draft release**, then do one of:<br>→ `/discard-snapshot <reason>` — discard and return to `planned`{{/state_snapshot_active}}{{#state_draft_ready}}<br>→ **`/publish-release --confirm {{release_tag}}` — publish the release**<br>→ `/delete-draft <reason>` — delete draft and return to `planned`{{/state_draft_ready}}{{^state_snapshot_active}}{{^state_draft_ready}}<br>→ **`/create-snapshot` — create a release snapshot**{{/state_draft_ready}}{{/state_snapshot_active}}
```

File: release_automation/templates/pr_bodies/release_review_pr.mustache [hunk 3]
```diff
@@ -1,25 +1,81 @@
...
+{{/is_stable_public}}
+
+### Release Management Review
+
+- [ ] CHANGELOG follows the release documentation rules
+- [ ] Breaking changes are documented and version updates follow SemVer rules
+- [ ] All mandatory readiness assets for the API versions are present and confirmed
+
+<details>
+<summary><b>Required assets per API status</b></summary>
```
Comment (tanjadegroot) [inline release_automation/templates/pr_bodies/release_review_pr.mustache]:
```suggestion
<summary><b>Required release assets per API status</b></summary>
```

**Review by tanjadegroot — COMMENTED** [review]
with propose changes

File: release_automation/docs/api-readiness-checklist.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
+# API Readiness Checklist
+
+This document defines the release assets that API teams must provide for each release, and how readiness is verified during the release process.
```
Comment (tanjadegroot) [inline release_automation/docs/api-readiness-checklist.md]:
```suggestion
This document defines the release assets that API teams must provide for each release depening on the API(s) status(es), and how readiness is verified during the release process.
```

File: release_automation/docs/api-readiness-checklist.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+### Release Management Review
+
+Release management reviewers verify process compliance:
+
+- CHANGELOG follows the release documentation rules
+- `release-metadata.yaml` content is correct
+- API version transformations were applied correctly (server URLs, references, version fields)
+- Version conventions are followed (SemVer, URL format matching API maturity)
+- All mandatory readiness assets for the declared release type are present
```
Comment (tanjadegroot) [inline release_automation/docs/api-readiness-checklist.md]:
```suggestion
- All mandatory release assets for the declared API(s) status(es) are present as per the checklist
```

File: release_automation/templates/pr_bodies/release_review_pr.mustache [hunk 6]
```diff
@@ -1,25 +1,81 @@
...
+| 7 | Test Cases (enhanced) | O | O | O | M |
+| 8 | API description link | O | O | M | M |
+
+M = Mandatory, O = Optional — [Full documentation](https://github.com/camaraproject/ReleaseManagement/blob/main/documentation/readiness/api-readiness-checklist.md)
+
+</details>
+
+**Verify snapshot content (during automation introduction phase):**
+- [ ] `release-metadata.yaml` content is correct
+- [ ] API version transformations are accurate (server URLs, references, version fields)
```
Comment (tanjadegroot) [inline release_automation/templates/pr_bodies/release_review_pr.mustache]:
```suggestion
- [ ] API version numbers and applicable extensions were generated correctly (server URLs, references, version fields)
```

File: release_automation/docs/api-readiness-checklist.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+| Release type | `repository.target_release_type` | `pre-release-alpha`, `pre-release-rc`, `public-release`, `maintenance-release` |
+| API status | `apis[].target_api_status` | `alpha`, `rc`, `public` |
+| Initial public | Derived | API status is `public` and major version is 0 (e.g., v0.5.0) |
+| Stable public | Derived | API status is `public` and major version ≥ 1 (e.g., v1.0.0) |
+
+The Release Issue and Release PR embed the readiness matrix as a convenience copy; this document is the canonical source.
+
+## Release Assets
+
+| Nr | Asset | Description | Location | Automated? | How to verify |
```
Comment (tanjadegroot) [inline release_automation/docs/api-readiness-checklist.md]:
```suggestion
| Nr | Asset | Description | Location | Automated? | What to verify |
```

File: release_automation/docs/api-readiness-checklist.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+| 4 | API Documentation | API description in the YAML `info.description` field; additional documentation as needed | In YAML `info` section or `documentation/` | Partial (`info.description` presence checked) | `info.description` is present and meaningful |
+| 5 | User Stories | At least one user story per API demonstrating the intended use | `documentation/API_documentation/` | No | At least one user story document per API exists |
+| 6 | Test Cases (basic) | Sunny day scenarios and main error cases; at least one `.feature` file per API | `code/Test_definitions/` | File existence only | At least one `.feature` file per API with sunny day + main error cases |
+| 7 | Test Cases (enhanced) | Rainy day scenarios, edge cases, and error handling coverage | `code/Test_definitions/` | No | Additional scenarios beyond basic: rainy day, edge cases, error handling |
+| 8 | API description link | Link to CAMARA Wiki API description page for external visibility | Wiki URL in API Readiness Checklist | No | Wiki URL is present and reachable |
```
Comment (tanjadegroot) [inline release_automation/docs/api-readiness-checklist.md]: For now, we could avoid hosting the URL.

File: release_automation/docs/api-readiness-checklist.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
+# API Readiness Checklist
+
+This document defines the release assets that API teams must provide for each release, and how readiness is verified during the release process.
+
+## Purpose
+
+Before an API repository can be released, codeowners must ensure that certain assets are in place and meet quality expectations appropriate for the declared target API status. This checklist:
+
+- Defines the required release assets and their expected locations
+- Specifies which assets are mandatory (M) or optional (O) per API status level
```
[the remark this answers is not in the document]
Comment (hdamker) [reply release_automation/docs/api-readiness-checklist.md]: Applied in https://github.com/camaraproject/tooling/pull/90/commits/f81dcdca781edd417df846b73d47c6b30aed00a0

File: release_automation/docs/api-readiness-checklist.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+| 5 | User Stories | O | O | O | M |
+| 6 | Test Cases (basic) | O | M | M | M |
+| 7 | Test Cases (enhanced) | O | O | O | M |
+| 8 | API description link | O | O | M | M |
+
+**Why this progression:**
+
+- **Alpha**: The API is under active development. Only the API definition and basic documentation are required. Teams are iterating on the design and gathering feedback.
+- **Release Candidate (rc)**: The API is feature-complete and ready for implementation testing. Commonalities compliance and basic test cases become mandatory to ensure interoperability.
+- **Initial Public**: The API is ready for first implementations by external parties. All API versions in the release are initial (major version 0.x). An API description link is added for external visibility.
```
[the remark this answers is not in the document]
Comment (hdamker) [reply release_automation/docs/api-readiness-checklist.md]: Applied in https://github.com/camaraproject/tooling/pull/90/commits/f81dcdca781edd417df846b73d47c6b30aed00a0. Removed the "link" in https://github.com/camaraproject/tooling/pull/90/commits/ba596ccb73651bb298d58a2ca1a5130658cd413b

File: release_automation/docs/api-readiness-checklist.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+- **CI checks are green**: Spectral linting and PR validation should pass on `main`
+- **All intended PRs are merged**: Implementation work should be complete on `main`
+- **SemVer is correct**: Breaking changes are only allowed in initial versions (v0.x) or new major versions
+- **Readiness assets are provided**: All mandatory assets for the declared target release type are in place (see matrix above)
+
+These items appear as a preparation checklist in the Release Issue while it is in PLANNED state. They are reminders, not automated gates — the codeowner is responsible for verifying them before proceeding.
+
+> **Note**: During development on `main`, API version fields in the YAML definitions must stay as `wip`. The release automation transformer replaces them with the correct version numbers during snapshot creation.
+
+## Review Process
```
[the remark this answers is not in the document]
Comment (hdamker) [reply release_automation/docs/api-readiness-checklist.md]: Applied in https://github.com/camaraproject/tooling/pull/90/commits/f81dcdca781edd417df846b73d47c6b30aed00a0
